### DISCUSSION TITLE
Can't get X and Y statement to work

### DISCUSSION THREAD

**Discussion by LilCelestial** [inkle/ink#774] [Writing in ink] [answered]
Hello, I'm desperate for help. I'm using inky for the first time. I've been trying for hours to get X and Y statements to work using variables I've set and I've had no luck. Can anyone see what I'm doing wrong?

This is what they look like:
{number_of_advice=3 and hasdate=1:->ConcludeDatePrep}
{(number_of_advice=3)&&(hasdate=1):->ConcludeDatePrep}
{ (number_of_advice=3) && (hasdate=1)}->ConcludeDatePrep

I've tried all the variations I can think of, > 2 instead of =3, the hasdate variable as true/false instead of numerical. My variables are set at the beginning of my file.

VAR number_of_advice = 0
VAR hasdate = 0

Any help would be hugely, massively, greatly appreciated!

**Comment by joningold** [comment, marked as answer]
Your ANDs are fine, but if you’re testing equals you need to use == not =.

A = B means “set A to the value of B” and can’t be used inside a condition
A == B means “compare A and B” and can’t be used outside of a condition

**Reply by LilCelestial** [reply]
Thank you Jon! I think I've incorrectly interpreted the use of AND statements. I thought I could use it for a statement in the vein of "if variable A is X and variable B is Y then: show this outcome". When you say 'compare A and B' what does that comparison mean/do?

I ended up creating a list and getting my desired result (in a more complex way). I have a bit more learning to do!

**Comment by joningold** [comment]
You can use an AND in that way. The problem was the use of a single equals
not a double equals.

Try copying this into inky:

~temp z= 1 // a single equals. This sets the value of z
{z==1} // this will say true; it’s comparing z and 1
{z==1 and z < 5} // this will say true
{z==2} // this will say false. It doesn’t set the value of z

**Reply by LilCelestial** [reply]
Thank you Jon, you've transformed what I can do. I really appreciate your help!
